From 94e1fdee8070b8232171bb0bf92bde2b21cac52f Mon Sep 17 00:00:00 2001
From: Roel Janssen <roel@gnu.org>
Date: Tue, 6 Mar 2018 10:01:49 +0100
Subject: [PATCH] hmf: hmf-pipeline: Add somatic PON filtering.

* lib/HMF/Pipeline/Config/Validate.pm: Split HMF_PON into HMF_GERMLINE_PON and
  HMF_SOMATIC_PON.
* settings/include/data.ini: Likewise.
* t/50-pipeline.pl.t: Likewise.
* templates/StrelkaPostProcess.sh.tt: Annotate and filter both somatic and
  germline PONs.
---
 lib/HMF/Pipeline/Config/Validate.pm |  3 ++-
 settings/include/data.ini           |  3 ++-
 templates/StrelkaPostProcess.sh.tt  | 33 +++++++++++++++++++++------------
 3 files changed, 25 insertions(+), 14 deletions(-)

diff --git a/lib/HMF/Pipeline/Config/Validate.pm b/lib/HMF/Pipeline/Config/Validate.pm
index 1607b9d..2292658 100644
--- a/lib/perl5/site_perl/5.26.0/HMF/Pipeline/Config/Validate.pm
+++ b/lib/perl5/site_perl/5.26.0/HMF/Pipeline/Config/Validate.pm
@@ -439,7 +439,8 @@ sub configChecks {
                 ANNOTATE_IDNAME => \&key_not_present,
                 ANNOTATE_IDDB => \&missing_file,
                 CALLING_DBSNP => \&missing_file,
-                HMF_PON => \&missing_file,
+                HMF_GERMLINE_PON => \&missing_file,
+                HMF_SOMATIC_PON => \&missing_file,
             }
         ),
         COPY_NUMBER => if_enabled({
diff --git a/settings/include/data.ini b/settings/include/data.ini
index 763bacc..99c6a06 100644
--- a/share/hmf-pipeline/settings/include/data.ini
+++ b/share/hmf-pipeline/settings/include/data.ini
@@ -27,7 +27,8 @@ ANNOTATE_FREQDB	/data/dbs/GoNL_v5/gonl.snps_indels.r5.sorted.vcf.gz
 ANNOTATE_IDNAME	Cosmicv76
 ANNOTATE_IDDB	/data/dbs/cosmic_v76/CosmicCodingMuts_v76.vcf.gz
 
-HMF_PON	/data/dbs/PON_v1.1/PON.vcf
+HMF_GERMLINE_PON	/data/dbs/PON_v1.1/GERMLINE_PON.vcf
+HMF_SOMATIC_PON	/data/dbs/PON_v1.1/SOMATIC_PON.vcf
 
 HIGH_CONFIDENCE_BED	/data/dbs/bed/NA12878_GIAB_highconf_IllFB-IllGATKHC-CG-Ion-Solid_ALLCHROM_v3.2.2_highconf.bed
 
diff --git a/templates/StrelkaPostProcess.sh.tt b/templates/StrelkaPostProcess.sh.tt
index f9e04d2..73fb695 100644
--- a/share/hmf-pipeline/templates/StrelkaPostProcess.sh.tt
+++ b/share/hmf-pipeline/templates/StrelkaPostProcess.sh.tt
@@ -69,27 +69,36 @@ java -Xmx[% opt.STRELKAPOSTPROCESS_MEM %]G \
 
 assert_last_position_unchanged "$input_vcf" "$output_vcf"
 [% opt.PBGZIP_PATH %]/pbgzip "$output_vcf"
-[% opt.TABIX_PATH %]/tabix "${output_vcf}.gz"
+[% opt.BCFTOOLS_PATH %]/bcftools index -tf "${output_vcf}.gz"
 
-rm "$input_vcf" "$input_vcf.idx"
+rm -f "$input_vcf" "$input_vcf.idx"
 input_vcf="${output_vcf}.gz"
-output_vcf="$basefile".annotated.pon.vcf
-rm -f "$output_vcf" "$output_vcf.idx"
+output_vcf="$basefile".germline.annotated.pon.vcf.gz
+rm -f "$output_vcf" "$output_vcf.tbi"
+
+[% opt.BCFTOOLS_PATH %]/bcftools annotate -a "[% opt.HMF_GERMLINE_PON %]" -c GERMLINE_PON_COUNT ${input_vcf} -o ${output_vcf} -O z
+[% opt.BCFTOOLS_PATH %]/bcftools index -tf "$output_vcf"
+
+rm -f "$input_vcf" "${input_vcf}.tbi"
+input_vcf="${output_vcf}"
+output_vcf="$basefile".germline.somatic.annotated.pon.vcf.gz
+rm -f "$output_vcf" "$output_vcf.tbi"
 
-[% opt.BCFTOOLS_PATH %]/bcftools annotate -a "[% opt.HMF_PON %]" -c PON_COUNT ${input_vcf} -o ${output_vcf}
+[% opt.BCFTOOLS_PATH %]/bcftools annotate -a "[% opt.HMF_SOMATIC_PON %]" -c SOMATIC_PON_COUNT ${input_vcf} -o ${output_vcf} -O z
+[% opt.BCFTOOLS_PATH %]/bcftools index -tf "$output_vcf"
 
 rm "$input_vcf" "$input_vcf.tbi"
 input_vcf="$output_vcf"
-output_vcf="$basefile".filtered.pon.vcf
-rm -f "$output_vcf" "$output_vcf.idx"
+output_vcf="$basefile".filtered.pon.vcf.gz
+rm -f "$output_vcf" "$output_vcf.tbi"
 
-[% opt.BCFTOOLS_PATH %]/bcftools filter -e 'PON_COUNT!="." && MIN(PON_COUNT) > 5' -s PON -m+ "$input_vcf" -o "$output_vcf"
-[% opt.IGVTOOLS_PATH %]/igvtools index "$output_vcf"
+[% opt.BCFTOOLS_PATH %]/bcftools filter -e 'GERMLINE_PON_COUNT!="." && MIN(GERMLINE_PON_COUNT) > 5' -s GERMLINE_PON -m+ "$input_vcf" -O u | \
+[% opt.BCFTOOLS_PATH %]/bcftools filter -e 'SOMATIC_PON_COUNT!="." && MIN(SOMATIC_PON_COUNT) > 3' -s SOMATIC_PON -m+ -o "$output_vcf" -O z
+[% opt.BCFTOOLS_PATH %]/bcftools index -tf "$output_vcf"
 
-assert_last_position_unchanged "$input_vcf" "$output_vcf"
-rm "$input_vcf"
+rm -f "$input_vcf"
 
 mv "$output_vcf" "[% final_vcf %]"
-mv "$output_vcf.idx" "[% final_vcf %].idx"
+mv "$output_vcf.tbi" "[% final_vcf %].tbi"
 
 success
-- 
2.16.1

